### PR TITLE
UIDATIMP-1656: Implement new Donor information section for Order field mapping profiles

### PR DESCRIPTION
**Pull request by mariia-aloshyna**
## Purpose

The data import field mapping profile for orders needs to be updated so that users can map data to the new donor information property rather than the deprecated donor field in the order record.

## Approach

- Add new section to the Orders edit screen
- Add new section to the Orders view screen

## Refs

https://folio-org.atlassian.net/browse/UIDATIMP-1656

### Edit screen
Screenshot 2024-08-20 at 22 36 14

### View screen
Screenshot 2024-08-20 at 22 36 33

### CODE REVIEW

**Review by OleksandrHladchenko1 — APPROVED** [review]

File: src/components/FieldDonor/FieldDonor.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  required = false,
+}) => {
+  const [selectedDonor, setSelectedDonor] = useState(null);
+  const [mappingQuery, setMappingQuery] = useState('');
+
+  useEffect(() => {
+    const uuid = getMatchByUuidInQuotes(inputValue);
+
+    if (uuid && allDonors?.length) {
+      if (selectedDonor?.id !== uuid) {
```
Comment (SKarolFolio) [inline src/components/FieldDonor/FieldDonor.js:37]:
```
if (uuid && allDonors?.length && selectedDonor?.id !== uuid) {
  ...
}
```

File: src/components/FieldDonor/FieldDonor.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    setSelectedDonor(donor);
+    setReferenceTables(name, `${mappingQuery}"${donor.id}"`);
+  };
+
+  const clearDonor = useCallback(() => {
+    setSelectedDonor(null);
+    setMappingQuery('');
+    setReferenceTables(name, '');
+
+    if (onClear) {
```
Comment (SKarolFolio) [inline src/components/FieldDonor/FieldDonor.js:64]: You can use optional chaining here:
`onClear?.()`

File: src/settings/MappingProfiles/detailsSections/edit/OrderDetailSection/DonorInformation.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  onAdd,
+  onRemove,
+} from '../../utils';
+import { DONORS_FIELD } from '../../../../../utils';
+
+export const DonorInformation = ({
+  initialFields,
+  setReferenceTables,
+}) => {
+  const DONORS_INDEX = 44;
```
Comment (SKarolFolio) [inline src/settings/MappingProfiles/detailsSections/edit/OrderDetailSection/DonorInformation.js:33]: Why not move `DONORS_INDEX` and `DONOR_INFORMATION_MAP` outside the component?

File: src/settings/MappingProfiles/detailsSections/view/OrderDetailsSection/DonorInformation.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      const orgId = getMatchByUuidInQuotes(value);
+      const organizationNameFromResponse = data?.find(item => item.id === orgId)?.name;
+
+      return organizationNameFromResponse ? value.replace(orgId, organizationNameFromResponse) : value;
+    }
+
+    return noValueElement;
+  } };
+
+  const donorsFieldsMap = [
```
Comment (SKarolFolio) [inline src/settings/MappingProfiles/detailsSections/view/OrderDetailsSection/DonorInformation.js:59]: same here

**Review by SKarolFolio — CHANGES_REQUESTED** [review]

File: src/settings/MappingProfiles/detailsSections/edit/OrderDetailSection/DonorInformation.js (hunk 3 above)
Comment (mariia-aloshyna) [reply src/settings/MappingProfiles/detailsSections/edit/OrderDetailSection/DonorInformation.js:33]: that's the common approach for all other sections

Comment (SKarolFolio) [reply src/settings/MappingProfiles/detailsSections/edit/OrderDetailSection/DonorInformation.js:33]: I mean, you can left it in the same file, but move outside the component.
Each time the component is re-rendered, this requires unnecessary calculations:
```
const DONOR_INFORMATION_MAP = {
  DONOR_NAME: index => getSubfieldName(DONORS_INDEX, 0, index),
};
```

**Review by SKarolFolio — APPROVED** [review]

**Pull request merged**